distributed/http/scheduler/api.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import json
+
+from distributed.http.utils import RequestHandler
+
+
+class APIHandler(RequestHandler):
+    def get(self):
+        self.write("API V1")
+        self.set_header("Content-Type", "text/plain")
+
+
+class RetireWorkersHandler(RequestHandler):
+    async def post(self):
+        self.set_header("Content-Type", "text/json")
+        scheduler = self.server
+        try:
+            params = json.loads(self.request.body)
+            workers_info = await scheduler.retire_workers(**params)
+            self.write(json.dumps(workers_info))
+        except Exception as e:
+            self.set_status(400, str(e))
+            self.write(json.dumps({"Error": "Bad request"}))
+
+
+class WorkersToCloseHandler(RequestHandler):
+    async def post(self):
+        self.set_header("Content-Type", "text/json")
+        scheduler = self.server
+        try:
+            params = json.loads(self.request.body)
+            workers_to_close = {"workers": scheduler.workers_to_close(**params)}
+            self.write(json.dumps(workers_to_close))
+        except Exception as e:
+            self.set_status(500, str(e))
+            self.write(json.dumps({"Error": "Bad request"}))
```

Comment:
1. This is not a `Bad request` but an internal server error now.
2. Why are we implementing an exception handler for this? Why not for all the request handlers? Isn't this default / 500 exception handler handled by `RequestHandler`, by default?
3. What happens to all other handlers if an exception occurs?

Reply:
Because it's taking arguments, and I wanted to handle that. So I guess we should include exception handling for all request handlers?